proxy/src/fully_qualified_authority.rs

```diff
@@ -120,13 +138,30 @@ mod tests {
             use http::uri::Authority;
 
             let input = Authority::from_shared(Bytes::from(input.as_bytes())).unwrap();
-            let output = super::FullyQualifiedAuthority::new(
-                &input, default_namespace, default_zone);
+            let output = match super::FullyQualifiedAuthority::new(
+                &input, default_namespace, default_zone) {
+                Some(output) => output,
+                None => panic!(
+                    "unexpected None for input={:?}, default_namespace={:?}, default_zone={:?}",
+                    input,
+                    default_namespace,
+                    default_zone
+                ),
+            };
             output.without_trailing_dot().into()
         }
 
-        assert_eq!("name",
-                   f("name", None, None));
+        fn none(input: &str, default_namespace: Option<&str>,
+             default_zone: Option<&str>) {
+            use bytes::Bytes;
+            use http::uri::Authority;
+
+            let input = Authority::from_shared(Bytes::from(input.as_bytes())).unwrap();
+            assert_eq!(None, super::FullyQualifiedAuthority::new(
+                &input, default_namespace, default_zone));
+        }
+
+        none("name", None, None);
```

Comment:
The tests show the effects, but I'll group them up here. These are all considered external names:

- `name` (no namespace, svc, or zone)
- `name.namespace.svc.cluster` (zone doesn't match `cluster.local`)
- `name.namespace.foo` (not svc)